account_invoice_discount_date/models/account_move.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2023 Moduon Team S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3.0)
+
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    discount_date = fields.Date(
+        compute="_compute_discount_date",
+        inverse="_inverse_discount_date",
+        store=True,
+        help="Last date at which the discounted amount must be paid in order "
+        "for the Early Payment Discount to be granted",
+    )
+
+    @api.depends("line_ids.discount_date")
+    def _compute_discount_date(self):
+        for record in self:
+            d_dates = sorted(
+                record.line_ids.filtered("discount_date").mapped("discount_date")
+            )
+            record.discount_date = d_dates and sorted(d_dates)[0] or None
```

Comment:
I think you don't need to sort them twice:

```suggestion
            record.discount_date = d_dates and d_dates[0] or None
```